Airwallex/Payment/Sources/View/Card/CardNumberTextField.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  CardNumberTextField.swift
+//  Airwallex
+//
+//  Created by Weiping Li on 2024/12/23.
+//  Copyright © 2024 Airwallex. All rights reserved.
+//
+
+import UIKit
+import Combine
+
+protocol CardNumberTextFieldConfiguring: ErrorHintableTextFieldConfiguring {
+    var supportedBrands: [AWXBrandType] { get }
+    var currentBrand: AWXBrandType { get }
+}
+
+class CardNumberTextField: BaseTextField {
+    
+    private let logoStack: UIStackView = {
+        let stack = UIStackView()
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.axis = .horizontal
+        stack.spacing = .spacing_4
+        for _ in 0..<4 {
+            let imageView = UIImageView()
+            imageView.translatesAutoresizingMaskIntoConstraints = false
+            imageView.contentMode = .scaleAspectFit
+            stack.addArrangedSubview(imageView)
+            
+            imageView.widthAnchor.constraint(equalToConstant: 30).isActive = true
+            imageView.heightAnchor.constraint(equalToConstant: 20).isActive = true
+        }
+        return stack
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        horizontalStack.addArrangedSubview(logoStack)
+        horizontalStack.setCustomSpacing(0, after: logoStack)
+        horizontalStack.addSpacer(.spacing_16)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func setup(_ viewModel: any BaseTextFieldConfiguring) {
+        super.setup(viewModel)
+        guard let viewModel = viewModel as? CardNumberTextFieldConfiguring else {
+            assert(false, "invalid view model type")
+            return
+        }
+        if viewModel.currentBrand != .unknown {
+            updateLogos(brands: [viewModel.currentBrand])
+        } else {
+            updateLogos(brands: viewModel.supportedBrands)
+        }
+    }
+    
+    private var timer: Timer?
```

Comment:
Do we need to invalidate it when view is deallocated? I think timer's lifecycle is attached to the runloop so might cause memory leak if not invalidate.